app/src/main/java/org/schabi/newpipe/RouterActivity.java

```diff
@@ -611,4 +644,103 @@ private String trim(final String input) {
+
+                    boolean isHTTPS = url.substring(0, 8).toLowerCase().equals("https://");
+                    Document page = Jsoup.parse(stream, "UTF-8", url);
+                    Elements metas = page.head().getElementsByTag("meta");
+
+                    for (int i = 0; i < metas.size(); i++) {
+                        Element meta = metas.get(i);
+
+                        String tmp = meta.attr("http-equiv");
+                        if (tmp.length() < 1) {
+                            continue;// find another tag
+                        }
+
+                        tmp = tmp.toLowerCase();
+                        if (tmp.equals("refresh") || tmp.equals("location")) {
+                            tmp = meta.attr("content");
+                            if (tmp.length() < 8) {
+                                continue;// content always MUST BE present
+                            }
+
+                            // read content attribute
+                            String[] pairs = tmp.split(";");// ¿by regex?
+
+                            int idx = pairs.length == 1 ? 0 : 1;
+                            pairs[idx] = pairs[idx].trim();
+                            if (pairs[idx].toUpperCase().startsWith("URL=")) {
+                                pairs[idx] = pairs[idx].substring(4);
+                            }
+
+                            /* Now check if the url is absolute. Valid cases:
+                             *       http://youtube.com/v?=abc123
+                             *       https://youtube.com/v?=abc123
+                             *       //youtube.com/v?=abc123
+                             */
+                            if (pairs[idx].length() < 8) {
+                                return null;// not a youtube link
+                            }
+                            tmp = pairs[idx].substring(0, 9).toLowerCase();
+
+                            if (tmp.startsWith("https://") || tmp.startsWith("http://")) {
+                                return pairs[idx];
+                            } else if (tmp.startsWith("//")) {
+                                // calculate relative protocol
+                                tmp = "http";
+                                if (isHTTPS) {
+                                    tmp = tmp.concat("s");
+                                }
+                                return tmp.concat(":").concat(pairs[idx]);
+                            }
+
+                            return null;// not valid url
+                        }
+                    }
+                    break;
+                case HttpURLConnection.HTTP_MOVED_TEMP:
+                case HttpURLConnection.HTTP_MOVED_PERM:
+                case HttpURLConnection.HTTP_SEE_OTHER:
+                    return con.getHeaderField("Location");
+            }
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to check for redirection in ".concat(url), e);
```

Comment:
We should use ErrorActivity not normal logs.

Reply:
is for debug only